pkg/pool-weighted/contracts/managed/ManagedPoolStorageLib.sol

```diff
@@ -45,13 +45,22 @@ library ManagedPoolStorageLib {
     uint256 private constant _SWAP_ENABLED_OFFSET = _SWAP_FEE_END_PCT_OFFSET + _SWAP_FEE_PCT_WIDTH;
     uint256 private constant _MUST_ALLOWLIST_LPS_OFFSET = _SWAP_ENABLED_OFFSET + 1;
     uint256 private constant _RECOVERY_MODE_OFFSET = _MUST_ALLOWLIST_LPS_OFFSET + 1;
+    uint256 private constant _JOIN_EXIT_ENABLED_OFFSET = _RECOVERY_MODE_OFFSET + 1;
 
     uint256 private constant _TIMESTAMP_WIDTH = 32;
     // 2**60 ~= 1.1e18 so this is sufficient to store the full range of potential swap fees.
     uint256 private constant _SWAP_FEE_PCT_WIDTH = 62;
 
     // Getters
 
+    /**
+     * @notice Returns whether the Pool allows regular joins and exits (recovery exits not included).
+     * @param poolState - The byte32 state of the Pool.
+     */
+    function getJoinsExitsEnabled(bytes32 poolState) internal pure returns (bool) {
```

Comment:
```suggestion
    function getJoinExitsEnabled(bytes32 poolState) internal pure returns (bool) {
```

We tend to use the term "joinExits" to refer to something which could be a join or an exit.